backend/src/main/scala/cromwell/backend/DuplicatingCacheHitCopyingActor.scala

```diff
@@ -0,0 +1,117 @@
+  protected def metadataKeyValues: Map[String, Any]
+
+  private def lookupSourceCallRootPath(sourceJobDetritusFiles: Map[String, String]): Path = {
+    sourceJobDetritusFiles.get(JobPaths.CallRootPathKey).map(getPath).getOrElse(throw new RuntimeException(
+      s"The call detritus files for source cache hit aren't found for call ${jobDescriptor.call.fullyQualifiedName}")
+    )
+  }
+
+  private def copySimpletons(wdlValueSimpletons: Seq[WdlValueSimpleton],
+                             sourceCallRootPath: Path): Unit = {
+    wdlValueSimpletons foreach {
+      case WdlValueSimpleton(key, wdlFile: WdlFile) =>
+        val sourcePath = getPath(wdlFile.value)
+        val destinationPath = PathCopier.getDestinationFilePath(sourceCallRootPath, sourcePath, destinationCallRootPath)
+        duplicate(sourcePath, destinationPath)
+      case _ => //ignoring non-WDLValueSimpleton:wdlFiles since they are not interesting here
+    }
+  }
+
+  private def copyDetritus(sourceJobDetritusFiles: Map[String, String]): Unit = {
+    val sourceKeys = sourceJobDetritusFiles.keySet
+    val destinationKeys = destinationJobDetritusPaths.keySet
+    val fileKeys = sourceKeys.intersect(destinationKeys).filterNot(_ == JobPaths.CallRootPathKey)
+
+    fileKeys foreach { fileKey =>
+      val sourcePath = getPath(sourceJobDetritusFiles(fileKey))
+      val destinationPath = destinationJobDetritusPaths(fileKey)
+      duplicate(sourcePath, destinationPath)
+    }
+  }
+
+  override def copyCachedOutputs(wdlValueSimpletons: Seq[WdlValueSimpleton],
+                                 sourceJobDetritusFiles: Map[String, String],
+                                 returnCodeOption: Option[Int]): BackendJobExecutionResponse = {
+    val sourceCallRootPath = lookupSourceCallRootPath(sourceJobDetritusFiles)
+
+    copySimpletons(wdlValueSimpletons, sourceCallRootPath)
+    copyDetritus(sourceJobDetritusFiles)
+
+    val sourceJobOutputs = WdlValueBuilder.toJobOutputs(jobDescriptor.call.task.outputs, wdlValueSimpletons)
+
+    val destinationJobOutputs = DuplicatingCacheHitCopyingActor.adjustFileOutputs(
+      sourceJobOutputs, sourceCallRootPath, destinationCallRootPath)
+
+    val jobDetritusFilesOption = Option(destinationJobDetritusPaths.mapValues(_.toString))
+
+    import cromwell.services.metadata.MetadataService.implicits.MetadataAutoPutter
+    serviceRegistryActor.putMetadata(jobDescriptor.workflowDescriptor.id, Option(jobDescriptor.key), metadataKeyValues)
+
+    SucceededResponse(jobDescriptor.key, returnCodeOption, destinationJobOutputs, jobDetritusFilesOption)
+  }
+}
+
+object DuplicatingCacheHitCopyingActor {
+
+  /**
+    * Adjusts the paths in jobOutputs from the sourceCallRoot to the destinationCallRoot.
+    */
+  def adjustFileOutputs(jobOutputs: JobOutputs, sourceCallRoot: Path, destinationCallRoot: Path) = {
+
+    def adjustWdlValue(wdlValue: WdlValue): WdlValue = wdlValue match {
+      case wdlFile: WdlFile if !wdlFile.isGlob => adjustWdlFile(wdlFile)
```

Comment:
Long story short: don't worry. An output will never be glob file type. 

That's only used in expression type evaluation when working out what to delocalise. By now they'll just be an Array[SingleFile]